app/components/UI/Ramp/deeplink/utils/getRedirectPathAndParams.ts

```diff
@@ -0,0 +1,19 @@
+const dummyProtocol = 'ramp:';
```

Comment:
Q: could you use another term than "dummy" ?
To me it means substitute for the real thing or placeholder data with the same format as real data but an actual one.
This is actually used. Not standardised outside of MM but not useless or placeholder either.

Reply:
It is just used to form a valid url for `URL` constructor. It is not used before of after. it can be anything `anything:`. It is not useless but is kind of a placeholder. What variable name do you suggest?

Reply:
Very good questions, let me explain:

You are correct, the `rampPath` string is **not** a URL, but it was part of one. This string is the result from the handler of deeplinks and universal links. For the [former](https://github.com/MetaMask/metamask-mobile/blob/7caff7bc06023086afc8c0d583d34ec5b213eb5b/app/core/DeeplinkManager/ParseManager/handleMetaMaskDeeplink.ts), the URL was [one of these](https://github.com/MetaMask/metamask-mobile/blob/7caff7bc06023086afc8c0d583d34ec5b213eb5b/app/core/DeeplinkManager/ParseManager/handleMetaMaskDeeplink.ts#L126-L142):

* `metamask://buy<rampPath>`
* `metamask://sell<rampPath>`
* `metamask://buy-crypto<rampPath>`
* `metamask://sell-crypto<rampPath>`

For the [latter](https://github.com/MetaMask/metamask-mobile/blob/feat/ramp-intent/app/core/DeeplinkManager/ParseManager/handleUniversalLink.ts), [one of these](https://github.com/MetaMask/metamask-mobile/blob/feat/ramp-intent/app/core/DeeplinkManager/ParseManager/handleUniversalLink.ts#L99-L108): 

* `https://metamask.app.link/buy<rampPath>`
* `https://metamask.app.link/buy-crypto<rampPath>`
* `https://metamask.app.link/sell<rampPath>`
* `https://metamask.app.link/sell-crypto<rampPath>`

>if it's not a URL but it's just handy to use the URL tools to split the string parts... then maybe you could encapsulate this into a type that may use URL internally but that would clarify that you do not expect this to directly be a URL.

This is what I wanted to express with `rampPath` type. it is just the raw `pathname` of the `URL` object resulting from combining any protocol with the ramp path (```new URL(`${dummyProtocol}${rampPath}`)```).

> Given the unit test, I see you use the params and path as in a URL, but at the same time there's all these /// at the begining that I don't really understand. So not really a URL. I would definitely favour a specific object to handle that even if its implementation uses URL under the hood for ease.

All of these `///` and weird cases come from the fact that we would be parsing the input from external links that might be malformed. Thankfully the `URL` handles that perfectly. 

My main question is, in more detail and concretly, what do you mean with "encapsulate this into a type" or "a specific object"? the `getRedirectPathsAndParams` itself is meant to be the function that handles this logic.

Reply:
in this case, if the protocol is really just dummy to make the URL work, then maybe simply:
```suggestion
const dummyProtocol = 'dummy:';
```

Reply:
works for me! Changed in a92b3be99